README.md

```diff
@@ -285,6 +285,73 @@ that match exclusive check.
 
 ```attr_pattern```, ```include``` and ```exclude``` arguments work on ```objprint```, ```objstr``` and ```@add_objprint```.
 
+### Register Custom Type Formatter
+
+You can also customize how certain types of objects are displayed by registering a custom formatter function to transform an object of a specific type into a string. For example, you can print all integers in hexadecimal format by registering the ```hex()``` function for the ```int``` data type, or registering a custom ```lambda``` function. You can always unregister the formatter function if you no longer want to use it with ```unregister_formatter()```, it supports unregistering multiple data types at the same time. And if no input argument is provided to the function, it will clear all the registered functions.
+
+```python
+from objprint import op
+
+op.register_formatter(int, hex)
+op.register_formatter(float, lambda x: f"{round(x, 3)}")
+op(10)  # prints 0xa
+op(3.14159)  # prints 3.142
+op.unregister_formatter(int, float)
+op(10)  # prints 10
+op(3.14159)  # prints 3.14159
+```
+
+Alternatively, you can also register a custom formatter function using a decorator:
+
+```python
+@op.register_formatter(str)
+def custom_formatter(obj: str):
+    return f"custom_print: {obj}"
+
+# This would clear all the registered formatters 
+op.unregister_formatter()
+```
+
+```Objprint``` always prioritizes custom formatters over default formatters and will use the registered functions if they exist. If type annotations are provided in the custom formatter functions, ```Objprint``` will also perform a validation check to ensure the provided function accepts the specified data type as input and returns a string.
+
+The ```op``` function call accepts an ```inherit``` argument for situations involving class inheritance. By setting ```inherit``` to ```True```, a formatter registered for a base class will also be applied to its derived classes, if the formatters for the derived classes are not registered. If a derived class inherits from multiple base classes with registered formatters, the choice of formatter follows the Method Resolution Order (MRO) of the derived class.
```

Comment:
Sorry I may not be clear on this. the ``inherit`` argument is not for ``op``, it's for ``register_formatter``. The user decides whether the formatter should be inherited when they register it,